File: services/changeowl-web/src/types/artifact-intelligence.ts
export type ChangeType =
  | "bugfix"
  | "feature"
  | "refactor"
  | "arch_change"
  | "chore";
export type ChangeSurface =
  | "api"
  | "runtime"
  | "dependency"
  | "internal"
  | "infra";
export type BehavioralImpact =
  | "none"
  | "bugfix"
  | "performance"
  | "functional_change"
  | "breaking_change";
export type RiskLevel = "low" | "medium" | "high";
export type Confidence = "high" | "medium" | "low";

export interface ArtifactIntelligence {
  rationale: string;
  intent: string;
  change_type: ChangeType;
  change_surface: ChangeSurface;
  behavioral_impact: BehavioralImpact;
  impact_radius: string[];
  key_points: string[];
  risk_level: RiskLevel;
  confidence: Confidence;
}

File: services/agent-service/app/models/artifact_intelligence.py
from typing import Literal

from pydantic import BaseModel


ChangeType = Literal[
    "bugfix",
    "feature",
    "refactor",
    "arch_change",
    "chore",
]

ChangeSurface = Literal[
    "api",
    "runtime",
    "dependency",
    "internal",
    "infra",
]

BehavioralImpact = Literal[
    "none",
    "bugfix",
    "performance",
    "functional_change",
    "breaking_change",
]

RiskLevel = Literal[
    "low",
    "medium",
    "high",
]

Confidence = Literal[
    "high",
    "medium",
    "low",
]


class ArtifactIntelligence(BaseModel):
    rationale: str
    intent: str

    change_type: ChangeType
    change_surface: ChangeSurface
    behavioral_impact: BehavioralImpact

    impact_radius: list[str]
    key_points: list[str]

    risk_level: RiskLevel
    confidence: Confidence